### DISCUSSION TITLE
Acesso à dados com .NET, C#, Dapper e SQL Server

### DISCUSSION THREAD

**Discussion by ThiagoSleme** [balta-io/forum#541] [Tira Dúvidas] [answered]
Boa noite, Balta! Tudo bem?

No acesso a dados com Dapper, quando rodo a aplicação consigo ver que o acesso ao banco foi realizado mas não retorna os dados da consulta no console porque da exception:

Unhandled exception. Microsoft.Data.SqlClient.SqlException (0x80131904): Uma conexão com o servidor foi estabelecida com êxito, mas ocorreu um erro durante o processo de logon. (provider: Provedor de SSL, error: 0 - A cadeia de certificação foi emitida por uma autoridade que não é de confiança.)
 ---> System.ComponentModel.Win32Exception (0x80090325): A cadeia de certificação foi emitida por uma autoridade que não é de confiança.
 
 consegue me ajudar ?

![2024-04-15_22h00_00](https://github.com/balta-io/forum/assets/31346179/fb47fd88-8148-4f58-a550-32b789c56c42)

**Comment by henriqueweege** [comment, marked as answer]
Oi, @ThiagoSleme ! Bom dia, tudo bem?
Meu caro, você poderia postar aqui um print da tua connection string?
Valeeeu!

**Comment by ThiagoSleme** [comment]
essa é a minha connection string:

const string connectionString = "Server=localhost,1433clear;Database=balta;User ID=sa;Password=Virtual0522!=@#01";

![image](https://github.com/balta-io/forum/assets/31346179/ccbf3ba2-ddf2-4f0a-aaef-71a7e89781a5)

**Reply by henriqueweege** [reply]
@ThiagoSleme , desculpa a demora! Você teria como testar com essa connectionString, por favor? "Server=localhost,1433;Database=balta;User ID=sa;Password=Virtual0522!=@#01; TrustServerCertificate=True; Trusted_Connection=True;"

Me avisa aqui se deu boa ;)

Valeeeu!

**Reply by ThiagoSleme** [reply]
não deu certo, apresenta o erro 

Unhandled exception. Microsoft.Data.SqlClient.SqlException (0x80131904): O nome da entidade de destino está incorreto. Não é possível gerar contexto SSPI.

**Reply by henriqueweege** [reply]
Oi, @ThiagoSleme! Então, esse erro não é o mesmo do erro anterior.
O problema é que não sei o que esse novo erro significa, mas não me parece ser algo no teu código e sim de configuração na tua máquina. Dei um Google aqui, e achei [esse artigo](https://learn.microsoft.com/pt-br/troubleshoot/sql/database-engine/connect/cannot-generate-sspi-context-error) e [esse artigo](https://learn.microsoft.com/pt-br/troubleshoot/windows-server/active-directory/target-principal-name-is-incorrect-when-replicating-data), mas nada que seja muito simples de resolver (como uma mudança na connection string, por exemplo).

Se puder abrir outro help com esse novo erro, te agradeço. Como o fórum funciona como pesquisa também, acho legal ter um novo para deixar mais fácil de alguém que venha a ter o mesmo problema encontrar a solução ;) 
Se puder, explica no novo help o que e o motivo do que foi feito aqui, só para contextualizar quem for te ajudar.

**Reply by lgc2** [reply]
@ThiagoSleme qualquer coisa tenta no modelo abaixo(trocando localhost por 127.0.0.1):

"Server=127.0.0.1,1433;Database=Blog;User ID=sa;Password=1q2w3e4r@#$;TrustServerCertificate=true");

Não se esqueça de substituir os dados.

**Reply by ThiagoSleme** [reply]
@lgc2  deu certo !
muito obrigado.

**Reply by lgc2** [reply]
@ThiagoSleme que bom, show de bola.

Apanhei um pouco com isso quando fui conectar no Azure DataStudio. A solução é simples, mas descobri la nem tanto.... Hahahah

**Comment by ThiagoSleme** [comment]
subi o código no meu repositório 

https://github.com/ThiagoSleme/Dapper_Balta.git
